unit_testing/DC/DC_tests.cpp

```diff
@@ -0,0 +1,506 @@
+/*!
+** @file   AC_tests.cpp
+** @author Luke W
+** @date   12/10/2023
+*/
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+/* Fakes */
+#include "fake_StmGpio.h"
+#include "fake_stm32xxxx_hal.h"
+#include "fake_USBprint.h"
+
+/* Real supporting units */
+#include "ADCmonitor.c"
+#include "CAProtocol.c"
+#include "CAProtocolStm.c"
+
+/* UUT */
+#include "DCBoard.c"
+
+using ::testing::AnyOf;
+using ::testing::AllOf;
+using ::testing::Contains;
+using ::testing::ElementsAre;
+using ::testing::IsEmpty;
+using namespace std;
+
+/***************************************************************************************************
+** TEST FIXTURES
+***************************************************************************************************/
+
+class DCBoard: public ::testing::Test 
+{
+    protected:
+        /*******************************************************************************************
+        ** METHODS
+        *******************************************************************************************/
+        DCBoard()
+        {
+            hadc.Init.NbrOfConversion = ACTUATIONPORTS;
+
+            /* OTP code to allow initialisation of the board to pass */
+            /* TODO: Make this use a define for the board release, so that tests fail if the someone
+            ** forgets to update the version numbers */
+            BoardInfo bi = {
+                .v2 = {
+                    .otpVersion = OTP_VERSION_2,
+                    .boardType  = DC_Board,
+                    .subBoardType = 0,
+                    .reserved = {0},
+                    .pcbVersion = {
+                        .major = 3,
```

Comment:
Should we go to the new format of the PCB version checking using pcbversion.sh and so on?